scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -283,8 +283,24 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
         }
     }
 
+    private fun getFileCount(result: JsonNode): Int {
+        result.get("headers")?.forEach { header ->
+            header.get("extra_data")?.get("files_count")?.let {
+                return it.intValue()
+            }
+        }
+
+        // Compute the value from scratch as fallback for backwards compatibility and to not
```

Comment:
Shouldn't the fallback simply be the original code before your change? Actually, the other way around: I'd check for `result["files_count"]` first as that's cheap, and use it if it exists. Otherwise use above code to get `files_count` from the header's extra data, and just fail if neither exists.

Reply:
Do you mean the fallback I implemented shall be removed ?

Reply:
Yes. I believe it's enough to look for `files_count` in those two places. We do not need to count files ourselves, and we didn't do that in the original code either.

Reply:
Done.